lms/djangoapps/instructor/tests/test_api.py

```diff
@@ -3856,6 +3858,118 @@ def test_show_student_extensions(self):
 
 
 @attr('shard_1')
+class TestCourseIssuedCertificatesData(SharedModuleStoreTestCase):
+    """
+    Test data dumps for issued certificates.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TestCourseIssuedCertificatesData, cls).setUpClass()
+        cls.course = CourseFactory.create()
+
+    def setUp(self):
+        super(TestCourseIssuedCertificatesData, self).setUp()
+        self.instructor = InstructorFactory(course_key=self.course.id)
+        self.client.login(username=self.instructor.username, password='test')
+
+    def generate_certificate(self, course_id, mode, status):
+        """
+        Generate test certificate
+        """
+        test_user = UserFactory()
+        GeneratedCertificateFactory.create(
+            user=test_user,
+            course_id=course_id,
+            mode=mode,
+            status=status
+        )
+
+    def test_certificates_features_against_status(self):
+        """
+        Test certificates with status 'downloadable' should be in the response.
+        """
+        url = reverse('get_issued_certificates_features', kwargs={'course_id': self.course.id.to_deprecated_string()})
+        # firstly generating downloadable certificates with 'honor' mode
+        certificate_count = 3
+        for __ in xrange(certificate_count):
+            self.generate_certificate(course_id=self.course.id, mode='honor', status=CertificateStatuses.generating)
+
+        response = self.client.get(url, {})
+        res_json = json.loads(response.content)
+        self.assertIn('certificates', res_json)
+        self.assertEqual(len(res_json['certificates']), 0)
+
+        # Certificates with status 'downloadable' should be in response.
+        self.generate_certificate(course_id=self.course.id, mode='honor', status=CertificateStatuses.downloadable)
+        response = self.client.get(url, {})
+        res_json = json.loads(response.content)
+        self.assertIn('certificates', res_json)
+        self.assertEqual(len(res_json['certificates']), 1)
+
+    def test_certificates_features_group_by_mode(self):
+        """
+        Test for certificate csv features against mode. Certificates should be group by 'mode' in reponse.
+        """
+        url = reverse('get_issued_certificates_features', kwargs={'course_id': self.course.id.to_deprecated_string()})
+        # firstly generating downloadable certificates with 'honor' mode
+        certificate_count = 3
+        for __ in xrange(certificate_count):
+            self.generate_certificate(course_id=self.course.id, mode='honor', status=CertificateStatuses.downloadable)
+
+        response = self.client.get(url, {})
+        res_json = json.loads(response.content)
+        self.assertIn('certificates', res_json)
+        self.assertEqual(len(res_json['certificates']), 1)
```

Comment:
I think there should be an assertion to check we have correct number of certificates.